asset/file_reader/slicer.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const { Slicer } = require('@terascope/job-components');
+const Queue = require('@terascope/queue');
+const { getOffsets } = require('@terascope/chunked-file-reader');
+const { TSError } = require('@terascope/utils');
+const Promise = require('bluebird');
+const path = require('path');
+const fse = require('fs-extra');
+
+class FileSlicer extends Slicer {
+    constructor(context, opConfig, executionConfig) {
+        super(context, opConfig, executionConfig);
+        this.client = fse;
+        this._queue = new Queue();
+        this._doneSlicing = false;
+    }
+
+    async initialize() {
+        this.checkProvidedPath(this.opConfig.path);
+        this.getFilePaths(this.opConfig.path)
```

Comment:
Since `this.getFilePaths` is being ran in the background we should make sure we handle the error to avoid `unhandledRejections`, additionally we not want to just "log" the error because if it does error we want the execution to fail. In reality we probably want this done chunk by chunk every time `slice` is called, so we don't overwhelm the filesystem or CPU.

Reply:
The reason for logging an error is to prevent an issue with a single file to hold up an entire job, but I did add a `.catch` on `this.getFilePaths(this.opConfig.path)`.

As for the slicing, this is a similar behavior to the S3 reader. Since there's no guarantee regarding file and/or subdirectory count in the provided `path`, I guess the underlying assumption here is that in adding or loading the files to be processed, a system running this job will already need to have the resources necessary to slice all the files.